pkgs/stdenv/generic/check-meta.nix

```diff
@@ -439,6 +450,18 @@ let
     # Please also update the type in /pkgs/top-level/config.nix alongside this.
     else if hasNoMaintainers attrs then
       { valid = "warn"; reason = "maintainerless"; errormsg = "has no maintainers"; }
+    else if unstructuredArrays attrs != [] then
+      let
+        quote = s: ''"${s}"'';
+        variables = unstructuredArrays attrs;
+        variablesStr = builtins.concatStringsSep ", " (map quote variables);
+        usePlural = variables != [(builtins.head variables)];
+        plural = optionalString usePlural "s";
+      in
+      { valid = "warn";
```

Comment:
I agree with @wolfgangwalther that this ought to be an error, but the set of packages it touches is pretty high. We'll patch it all up in the next staging-next.

Reply:
We need to consider out‐of‐tree users too. I agree with the suggestion to add the “this will break” warning and then make it a hard error on the next release.